### PR TITLE
#166789990 User Should Be Able To Receive Verification Email After Signup

### PR DESCRIPTION
**Pull request by IsaiahRn**
#### What does this PR do?

- Verify user account

#### Description of Task to be completed

- User should be able to receive a verification link via email upon successful signup.

#### How should this be manually tested?

1. Run the following commands in Terminal

```
git clone git@github.com:andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-verification-email-166789990
npm install
npm start
```

2. Add environment variables as specified in `.env.sample` file

3. Go to the postman and Register a user via `http://localhost:3000/api/v1/signup`, 

4. A verification email will be sent to the user after successfully signing up.

5. Go to your email and verify user account via a button `Verify Account`.

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[166789990](https://www.pivotaltracker.com/story/show/166789990)
#### Screenshots (if appropriate)

- After successful registration, the user should receive a verification email.
Screen Shot 2019-07-07 at 19 12 33

- Through the email, user should click on `Verify Account` button.
Screen Shot 2019-07-06 at 21 52 13

- Finally, Account will be verified.
Screen Shot 2019-07-06 at 21 52 46

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
kindly follow the endpoints in the `readme.md` file, it should be `/api/` not `/api/v1`

File: controllers/auth.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+export const signup = async (req, res) => {
+  const {
+    username, firstname, lastname, email, password,
+  } = req.body;
+  const hashedPassword = Auth.hashPassword(password);
+  const transaction = await db.sequelize.transaction();
+  try {
+    const newUser = await db.users.create({
```
Comment (kodek-sleuth) [inline controllers/auth.js]: destructure all your objects where possible

**Review by kodek-sleuth — APPROVED** [review]

**Review by niyoceles — DISMISSED** [review]

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]
Please use object restructuring where ever possible

**Comment by elemanhillary-zz** [comment]
> Please use object restructuring where ever possible

**Review by Hervera — DISMISSED** [review]

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
squash your commit messages since they have exceeded the required number of messages a PR should have

**Review by Hervera — APPROVED** [review]

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

**Review by kodek-sleuth — APPROVED** [review]
This pr is ready to be merged

**Review by niyoceles — APPROVED** [review]

**Pull request merged**